src/engine/BMInterfaceTournament.php

```diff
@@ -0,0 +1,987 @@
+                                ':description' => $description);
+            self::$db->update($query, $parameters);
+
+            $tournamentId = self::$db->select_single_value('SELECT LAST_INSERT_ID()', array(), 'int');
+
+            return $tournamentId;
+        } catch (BMExceptionDatabase $e) {
+            $this->set_message('Cannot insert tournament because the creator ID was not valid');
+            return NULL;
+        } catch (Exception $e) {
+            // Failure might occur on DB insert or afterward
+            $errorData = $statement->errorInfo();
+            if ($errorData[2]) {
+                $this->set_message('Tournament create failed: ' . $errorData[2]);
+            } else {
+                $this->set_message('Tournament create failed: ' . $e->getMessage());
+            }
+            error_log(
+                'Caught exception in BMInterface::insert_new_tournament: ' .
+                $e->getMessage()
+            );
+            return NULL;
+        }
+    }
+
+    /**
+     * Load a tournament from the database
+     *
+     * @param int $tournamentId
+     * @return NULL|BMTournament
+     */
+    protected function load_tournament($tournamentId) {
+        try {
+            $tournament = $this->load_tournament_parameters($tournamentId);
+
+            // check whether the game exists
+            if (!isset($tournament)) {
+                $this->set_message("Tournament $tournamentId does not exist.");
+                return NULL;
+            }
+
+            if ('' == $this->message) {
+                $this->set_message("Loaded data for tournament $tournamentId.");
+            }
+
+            return $tournament;
+        } catch (BMExceptionDatabase $e) {
+            $this->set_message('Cannot load tournament the tourn ID was not valid');
+            return NULL;
+        } catch (Exception $e) {
+            error_log(
+                'Caught exception in BMInterface::load_tournament: ' .
+                $e->getMessage()
+            );
+            $this->set_message("Internal error while loading tournament.");
+            return NULL;
+        }
+    }
+
+    /**
+     * Most of the tournament loading logic
+     *
+     * @param int $tournamentId
+     * @return NULL|BMTournament
+     */
+    protected function load_tournament_parameters($tournamentId) {
+        $query = 'SELECT t.tournament_state,'.
+                 't.round_number,'.
+                 't.n_players,'.
+                 't.n_target_wins,'.
+                 't.tournament_type,'.
+                 't.creator_id,'.
+                 't.description,'.
+                 'g.id AS game_id,'.
+                 'g.tournament_round_number AS tournament_round_number '.
+                 'FROM tournament AS t '.
+                 'LEFT JOIN game AS g '.
+                 'ON t.id = g.tournament_id '.
+                 'WHERE t.id = :tournament_id '.
+                 'ORDER BY g.id;';
```

Comment:
Why does this work?  If a tournament doesn't have any games yet, will it actually return any rows here?

I'm probably just misreading the MySQL, but i wanted to check.

Reply:
Yup, the LEFT JOIN takes care of that.